statsmodels/tsa/statespace/sarimax.py

```diff
@@ -0,0 +1,1941 @@
+    k_seasonal_ma_params : int
+        Number of seasonal moving average parameters to be estimated.
+    k_trend : int
+        Order of the trend polynomial plus one (i.e. the constant polynomial
+        would have `k_trend=1`).
+    k_exog : int
+        Number of exogenous regressors.
+
+    Notes
+    -----
+    The SARIMA model is specified :math:`(p, d, q) \times (P, D, Q)_s`.
+
+    .. math::
+
+        \phi_p (L) \tilde \phi_P (L^s) \Delta^d \Delta_s^D y_t = A(t) +
+            \theta_q (L) \tilde \theta_Q (L^s) \zeta_t
+
+    In terms of a univariate structural model, this can be represented as
+
+    .. math::
+
+        y_t = u_t + \eta_t \\
+        \phi_p (L) \tilde \phi_P (L^s) \Delta^d \Delta_s^D u_t = A(t) +
+            \theta_q (L) \tilde \theta_Q (L^s) \zeta_t
+
+    where :math:`\eta_t` is only applicable in the case of measurement error
+    (although it is also used in the case of a pure regression model, i.e. if
+    p=q=0).
+
+    In terms of this model, regression with SARIMA errors can be represented
+    easily as
+
+    .. math::
+
+        y_t = \beta_t x_t + u_t \\
+        \phi_p (L) \tilde \phi_P (L^s) \Delta^d \Delta_s^D u_t = A(t) +
+            \theta_q (L) \tilde \theta_Q (L^s) \zeta_t
+
+    this model is the one used when exogenous regressors are provided.
+
+    Note that the reduced form lag polynomials will be written as:
+
+    .. math::
+
+        \Phi (L) \equiv \phi_p (L) \tilde \phi_P (L^s) \\
+        \Theta (L) \equiv \theta_q (L) \tilde \theta_Q (L^s)
+
+    If `mle_regression` is True, regression coefficients are treated as
+    additional parameters to be estimated via maximum likelihood. Otherwise
+    they are included as part of the state with a diffuse initialization.
+    In this case, however, with approximate diffuse initialization, results
+    can be sensitive to the initial variance.
+
+    This class allows two different underlying representations of ARMA models
+    as state space models: that of Hamilton and that of Harvey. Both are
+    equivalent in the sense that they are analytical representations of the
+    ARMA model, but the state vectors of each have different meanings. For
+    this reason, maximum likelihood does not result in identical parameter
+    estimates and even the same set of parameters will result in different
+    loglikelihoods.
+
+    The Harvey representation is convenient because it allows integrating
+    differencing into the state vector to allow using all observations for
+    estimation.
+
+    In this implementation of differenced models, the Hamilton representation
+    is not able to accomodate differencing in the state vector, so
+    `simple_differencing` (which performs differencing prior to estimation so
+    that the first d + sD observations are lost) must be used.
+
+    Many other packages use the Hamilton representation, so that tests against
+    Stata and R require using it along with simple differencing (as Stata
+    does).
+
```

Comment:
The main point to write a "full story" for you would be if you want to publish an article about it.

One helpful hint for us, statsmodels developers and users, are also the relevant references, for example 
Are the different options and state-space representation all in the Durbin Koopman book?
A list with reference to specific sections or chapters would also be helpful, if available.
e.g. time-varying coefficients follows the implementation in section xxx for mle_regression and section xxy for statespace regression.